pkg/client/client_test.go

```diff
@@ -391,6 +393,127 @@ var _ = Describe("Client", func() {
 		})
 	})
 
+	Describe("StatusClient", func() {
+		It("should update status of an existing object", func(done Done) {
+			cl, err := client.New(cfg, client.Options{})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(cl).NotTo(BeNil())
+
+			By("initially creating a Deployment")
+			dep, err := clientset.AppsV1().Deployments(ns).Create(dep)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("updating the status of Deployment")
+			dep.Status.Replicas = 1
+			err = cl.Status().Update(context.TODO(), dep)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("validating updated Deployment has new status")
+			actual, err := clientset.AppsV1().Deployments(ns).Get(dep.Name, metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(actual).NotTo(BeNil())
+			Expect(actual.Status.Replicas).To(BeEquivalentTo(1))
+
+			close(done)
+		})
+
+		It("should not update spec of an existing object", func(done Done) {
+			cl, err := client.New(cfg, client.Options{})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(cl).NotTo(BeNil())
+
+			By("initially creating a Deployment")
+			dep, err := clientset.AppsV1().Deployments(ns).Create(dep)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("updating the spec and status of Deployment")
+			var rc int32 = 1
+			dep.Status.Replicas = 1
+			dep.Spec.Replicas = &rc
+			err = cl.Status().Update(context.TODO(), dep)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("validating updated Deployment has new status and unchanged spec")
+			actual, err := clientset.AppsV1().Deployments(ns).Get(dep.Name, metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(actual).NotTo(BeNil())
+			Expect(actual.Status.Replicas).To(BeEquivalentTo(1))
+			Expect(*actual.Spec.Replicas).To(BeEquivalentTo(replicaCount))
+
+			close(done)
+		})
+
+		It("should update an existing object non-namespace object", func(done Done) {
+			cl, err := client.New(cfg, client.Options{})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(cl).NotTo(BeNil())
+
+			node, err := clientset.CoreV1().Nodes().Create(node)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("updating status of the object")
+			node.Status.Phase = corev1.NodeRunning
+			err = cl.Status().Update(context.TODO(), node)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("validate updated Node had new annotation")
+			actual, err := clientset.CoreV1().Nodes().Get(node.Name, metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(actual).NotTo(BeNil())
+			Expect(actual.Status.Phase).To(Equal(corev1.NodeRunning))
+
+			close(done)
+		})
+
+		It("should fail if the object does not exists", func(done Done) {
+			cl, err := client.New(cfg, client.Options{})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(cl).NotTo(BeNil())
+
+			By("updating status of a non-existent object")
+			err = cl.Status().Update(context.TODO(), dep)
+			Expect(err).To(HaveOccurred())
+
+			close(done)
+		})
+
+		It("should fail if the object does not pass server-side validation", func() {
```

Comment:
I don't think we need to explicitly test server-side behavior.  The implication is that we're using the status subresource here.

Reply:
+1, removed it.